### PR TITLE
Better separation of AidRequests and the map logic that is independent of it

### PR DESCRIPTION
**Pull request by AntonRaichuk**

### CODE REVIEW

**Review by jciepka — APPROVED** [review]

File: src/pages/inventory/index.tsx [hunk 1]
```diff
@@ -41,7 +41,7 @@ export function Inventory() {
 
   const { addFilter, getActiveFilterItems } = filterContext;
 
-  // First create a lookup table for all aid requests grouped by dates and memoise it
+  // First create a lookup table for all stock items grouped by dates and memoise it
```
Comment (MaciejReimann) [inline src/pages/inventory/index.tsx:44]: **issue**

Why this change? `AidRequest` was / is a data model designed to conform to: 
- business requirements
- BE data structure

I have no idea, what a stock item is and where it is coming from.

Besides, an AidRequest can have multiple stock items, I guess.

File: src/others/helpers/map-utils.ts [hunk 2]
```diff
@@ -50,3 +27,34 @@ export const mapAidRequestsToFeatures = (mapData: RequestMapDataPoint[], locatio
   });
 };
 
+export type RegionMetadata = { [id: string]: {amount: number, description: string } };
```
Comment (MaciejReimann) [inline src/others/helpers/map-utils.ts:30]: **issue/naming**
Why `Metadata`? It's just exactly the same data, but `GroupedByRegion`, isn't it? 
I find such inconsistencies extremely confusing.

File: src/others/helpers/map-utils.ts [hunk 3]
```diff
@@ -50,3 +27,34 @@ export const mapAidRequestsToFeatures = (mapData: RequestMapDataPoint[], locatio
...
+type RegionData = {
+  region_id: string;
+  city_name: string;
+  amount: number;
+};
+
+export const mapRegionIdsToMetadata = (mapDataPoints: MapDataPoint[], translateLocation: (city_id: number) => Location | undefined): RegionMetadata => {
```
Comment (MaciejReimann) [inline src/others/helpers/map-utils.ts:37]: **issue / naming**

If you decide to stick to the naming schema already in place, this would be called `groupByRegions`

File: src/others/helpers/map-utils.ts [hunk 4]
```diff
@@ -50,3 +27,34 @@ export const mapAidRequestsToFeatures = (mapData: RequestMapDataPoint[], locatio
...
+    if (!city) throw new Error(`Loccation ${req.city_id} is not found`);
+    return {
+      region_id: city.region_id,
+      city_name: city.name,
+      amount: req.amount,
+    };
+  });
+  const groupedRegionData = Object.entries(groupBy(regionData, "region_id"));
+  const regionToMetadata: RegionMetadata = {};
+  groupedRegionData.forEach(([region_id, data]) => {
```
Comment (MaciejReimann) [inline src/others/helpers/map-utils.ts:49]: **issue**
I don't really understand, why use `forEach` if we could easily use 2 lodash functions and make this code much clearer.

**Review by MaciejReimann — CHANGES_REQUESTED** [review]

File: src/pages/inventory/index.tsx (hunk 1 above)
Comment (MaciejReimann) [reply src/pages/inventory/index.tsx:44]: Ok, sorry, it's inventory, not requests. My bad!

**Review by MaciejReimann — APPROVED** [review]

File: src/others/helpers/aid-request-helpers.test.ts [hunk 5]
```diff
@@ -181,37 +181,19 @@ test("groupByCategoryIdWithTotal with exampleAggregatedRequests", () => {
...
-      description: "name#3: 50\nname#1: 10\n",
-    },
+  function mockSupplyTranslator(category_id: string) {
+    return {id: category_id, name: `Name#${category_id}`};
+  }
+
+  expect(aggregateCategories({city_id: 1, total: 17, aidRequests: requests}, mockSupplyTranslator)).toEqual({
+    amount: 17,
+    city_id: 1,
+    description: "Name#water: 20\nName#personal_hygiene_kits: 14\nName#food: 14\n",
```
[the remark this answers is not in the document]
Comment (AntonRaichuk) [reply src/others/helpers/aid-request-helpers.test.ts:197]: Ack. We have a TODO in the real code for this.

File: src/others/helpers/map-utils.ts (hunk 4 above)
Comment (MaciejReimann) [reply src/others/helpers/map-utils.ts:49]: I really don't believe you're a noob :D

It's not about styleguides, the way I see it at least. It's about that you listed 3 things that are happening there at once, while in an ideal world I'd see one at a time. 

The only style-related issue is that you're mutating data, which we aim at avoiding, that's why we use map/filter/reduce and all the derivatives instead.

Comment (AntonRaichuk) [reply src/others/helpers/map-utils.ts]: TypeScript - noob.

Done.
Found a way how to create a dict with a map, so changed forEach with a map and now the inside computation does not change anything outside its scope.

File: src/others/helpers/map-utils.ts (hunk 2 above)
Comment (AntonRaichuk) [reply src/others/helpers/map-utils.ts:30]: You mean the same data as MapDataPoint?

The two differences are:
 1) city_id -> region_id
 2) MapDataPoint is used for lists, while this one defines a dict.

So what is your exact proposal to name this type? 

If `GroupedByRegion` then i don't really like it (my not liking applies to existing names too).
The reason is GroupedBy* describes how the data was obtained and not what data does the object represent.
And I think the types should describe the data being stored, not how it was obtained.

For variable names it makes sense to have "groupedBy*" as part of the name, but not the type.

wdyt?

File: src/others/helpers/map-utils.ts (hunk 3 above)
Comment (AntonRaichuk) [reply src/others/helpers/map-utils.ts]: done
